core/src/main/java/hudson/FilePath.java

```diff
@@ -3322,6 +3324,19 @@ public Boolean invoke(@Nonnull File parentFile, @Nonnull VirtualChannel channel)
             }
             return current;
         }
+        
+        private Path relaxedToRealPath(Path path) throws IOException {
+            try {
+                return path.toRealPath();
+            }
+            catch(IOException e) {
+                LOGGER.log(Level.FINE, String.format("relaxedToRealPath cannot use the regular toRealPath on %s, trying with toRealPath(LinkOption.NOFOLLOW_LINKS)", path), e);
```

Comment:
You should use the lambda form here instead: `LOGGER.fine(() -> String.format(...))`

Reply:
You mean, like nowhere else in the project? ;) 

I am not convinced that the lambda will help in terms of performance or if it helps, in a visible way.

Reply:
The lambda form was added in Java 8. Jenkins is older than the Earth itself I think.

Anyways, the alternative is to surround it in `if (LOGGER.isLoggable(Level.FINE)) { ... }` to avoid all that string concatenation/construction when logging is disabled there. I could see that being an issue if this method is called on every file in an archive, for example.

Reply:
I added the `isLoggable`. 

I am still not convinced of the performance impact as we are already in a catch, meaning there was an exception creation that is the one of the most expensive operation in an application outside IO. Especially with `String.format` that has a complexity linear in the length of the String. From my PoV it's like adding a `O(1)` operation at the end of a `O(n)` algorithm, there is no real impact.

Reply:
Log messages add up, and it's like one of the biggest performance complaints in typical applications.